src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -0,0 +1,114 @@
+    public sealed class SonarQubeIssuesProvider : ISonarQubeIssuesProvider, IDisposable
+    {
+        private const double MillisecondsToWaitBetweenRefresh = 1000 * 60 * 60 * 1; // 1 hour
+
+        private readonly System.Timers.Timer refreshTimer;
+        private readonly IActiveSolutionBoundTracker solutionBoundTacker;
+        private readonly ISonarQubeService sonarQubeService;
+
+        private IList<SonarQubeIssue> cachedSuppressedIssues;
+        private bool isDisposed;
+        private CancellationTokenSource cancellationTokenSource;
+
+        [ImportingConstructor]
+        internal SonarQubeIssuesProvider(ISonarQubeService sonarQubeService, IActiveSolutionBoundTracker solutionBoundTacker)
+        {
+            this.sonarQubeService = sonarQubeService;
+            this.solutionBoundTacker = solutionBoundTacker;
+            this.solutionBoundTacker.SolutionBindingChanged += OnSolutionBoundChanged;
+
+            // TODO: Use mockable timer
+            refreshTimer = new System.Timers.Timer { AutoReset = true, Interval = MillisecondsToWaitBetweenRefresh };
+            refreshTimer.Elapsed += OnRefreshTimerElapsed;
+
+            if (this.solutionBoundTacker.IsActiveSolutionBound)
+            {
+                SynchronizeSuppressedIssues();
+                refreshTimer.Start();
+            }
+        }
+
+        public void Dispose()
+        {
+            if (this.isDisposed)
+            {
+                return;
+            }
+
+            refreshTimer.Dispose();
+            this.solutionBoundTacker.SolutionBindingChanged -= OnSolutionBoundChanged;
+            this.isDisposed = true;
+        }
+
+        public IEnumerable<SonarQubeIssue> GetSuppressedIssues(string projectGuid, string filePath)
+        {
+            // TODO: Block the call while the cache is being built + handle multi-threading
+            return this.cachedSuppressedIssues.Where(x => x.FilePath == filePath && x.ModuleKey == BuildModuleKey(projectGuid));
+        }
+
+        private string BuildModuleKey(string projectGuid)
+        {
+            return $"{solutionBoundTacker.ProjectKey}:{solutionBoundTacker.ProjectKey}:{projectGuid}";
+        }
+
+        private async void OnRefreshTimerElapsed(object sender, System.Timers.ElapsedEventArgs e)
+        {
+            await SynchronizeSuppressedIssues();
+        }
+
+        private async void OnSolutionBoundChanged(object sender, ActiveSolutionBindingEventArgs eventArgs)
+        {
+            if (solutionBoundTacker.IsActiveSolutionBound)
+            {
+                await SynchronizeSuppressedIssues();
+                refreshTimer.Start();
+            }
+            else
+            {
+                refreshTimer.Stop();
+            }
+        }
+
+        private async Task SynchronizeSuppressedIssues()
+        {
+            cancellationTokenSource?.Cancel();
+            cancellationTokenSource = new CancellationTokenSource();
+
+            // TODO: Handle race conditions
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)